polkadb/database.go

```diff
@@ -89,6 +94,8 @@ func (db *BadgerDB) Put(key []byte, value []byte) error {
 
 // Has checks the given key exists already; returning true or false
 func (db *BadgerDB) Has(key []byte) (exists bool, err error) {
+	db.lock.Lock()
+	defer db.lock.Unlock()
```

Comment:
I think you want `bs.lock.RLock()` and `bs.lock.RUnlock` for read-only ops

Reply:
oh right, good point